### PR TITLE
Add `jerry_string_to_utf8_char_buffer` API function.

### PR DESCRIPTION
**Pull request by robertsipka**
Copy the characters of a string into a specified utf-8 string buffer.

### CODE REVIEW

File: jerry-core/jerry.c [hunk 1]
```diff
@@ -1160,10 +1160,44 @@ jerry_string_to_char_buffer (const jerry_value_t value, /**< input string value
...
+  return ecma_string_copy_to_cesu8_buffer (str_p,
+                                           (lit_utf8_byte_t *) buffer_p,
+                                           buffer_size);
+} /* jerry_string_to_char_buffer */
+
+/**
+ * Copy the characters of an utf-8 encoded string into a specified buffer.
+ *
+ * Note:
+ *      The '\0' character could occur in character buffer.
```
Comment (zherczeg) [inline jerry-core/jerry.c]: could occur anywhere in the returned string

File: jerry-core/lit/lit-strings.c [hunk 2]
```diff
@@ -779,6 +779,57 @@ lit_code_point_to_utf8 (lit_code_point_t code_point, /**< code point */
 } /* lit_code_point_to_utf8 */
 
 /**
+ * Convert cesu-8 string to an utf-8 string and put it to the buffer.
```
Comment (zherczeg) [inline jerry-core/lit/lit-strings.c]: put it into

File: jerry-core/lit/lit-strings.c [hunk 3]
```diff
@@ -779,6 +779,57 @@ lit_code_point_to_utf8 (lit_code_point_t code_point, /**< code point */
...
+  ecma_char_t prev_ch = 0;
+
+  while (cesu8_pos < cesu8_end_pos)
+  {
+    ecma_char_t ch;
+    lit_utf8_size_t code_unit_size = lit_read_code_unit_from_utf8 (cesu8_pos, &ch);
+
+    if (lit_is_code_point_utf16_low_surrogate (ch) && lit_is_code_point_utf16_high_surrogate (prev_ch))
+    {
+      utf8_pos -= code_unit_size;
```
Comment (zherczeg) [inline jerry-core/lit/lit-strings.c]: This is quite clever idea.

One more thing: the cesu8 size of low and high surrogate are equals so substracting code_unit_size works, but an assert could be good here.

Comment (robertsipka) [inline jerry-core/lit/lit-strings.c]: Thanks, I've modified this patch, based on your suggestions.

File: tests/unit/test-api.c [hunk 4]
```diff
@@ -360,6 +360,26 @@ main (void)
   jerry_release_value (args[0]);
   jerry_release_value (args[1]);
 
+  /* Test jerry_string_to_utf8_char_buffer, test string: 'str: {DESERET CAPITAL LETTER LONG I}' */
+  args[0] = jerry_create_string_from_utf8 ((jerry_char_t *) "\x73\x74\x72\x3a \xf0\x90\x90\x80");
+  args[1] = jerry_create_string ((jerry_char_t *) "\x73\x74\x72\x3a \xed\xa0\x81\xed\xb0\x80");
+
+  /* these size must be equal */
```
Comment (zherczeg) [inline tests/unit/test-api.c]: Thise sizes must be equal.

Comment (LaszloLango) [inline tests/unit/test-api.c]: *These

**Comment by robertsipka** [comment]
Thanks, I've fixed the comments, and rebased it with master.

**Pull request merged**